Automattic/wp-calypso
# My Email: Add/add mailbox selection screen

#### Changes proposed in this Pull Request

* Adds a component that lists all the mailboxes across all domains in the site
* The mailbox items are fetched via the `useGetMailboxes` query
* Adds responsive views for both desktop and mobile scenarios
* Adds a prompt to add a new mailbox
* Includes a progress loader that provides interaction while items are being loaded


#### Testing instructions
- Checkout this branch
- Have a site with at least one domain, and one or more mailboxes per domain i.e. at least one email subscription
- Navigate to `/inbox/<site-slug>` . 
- Confirm the following screens



<img width="500" alt="Screenshot 2021-09-22 at 12 55 16 PM" src="https://user-images.githubusercontent.com/277661/134339090-90587e26-3d8c-48f6-a95d-7fa8d464d302.png">


<img width="500" alt="Screenshot 2021-09-22 at 12 55 04 PM" src="https://user-images.githubusercontent.com/277661/134339132-056ed54d-8b68-4da4-af68-29b4e6fd8dc5.png">




*
